src/backend/api/Fusion.Resources.Domain/Commands/Requests/Sharing/ShareRequest.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Fusion.Resources.Database;
+using Fusion.Resources.Database.Entities;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Fusion.Resources.Domain.Commands.Requests.Sharing
+{
+    public class ShareRequest : TrackableRequest<bool>
+    {
+        public ShareRequest(Guid requestId, string scope, string source, string? reason = null)
+        {
+            RequestId = requestId;
+            Scope = scope;
+            Source = source;
+            Reason = reason;
+        }
+
+        public Guid RequestId { get; set; }
+        public List<PersonId> SharedWith { get; set; } = new();
+        public string Scope { get; }
+        public string Source { get; }
+        public string? Reason { get; }
+
+        public class Handler : IRequestHandler<ShareRequest, bool>
+        {
+            private readonly ResourcesDbContext db;
+            private readonly IProfileService profileService;
+
+            public Handler(ResourcesDbContext db, IProfileService profileService)
+            {
+                this.db = db;
+                this.profileService = profileService;
+            }
+
+            public async Task<bool> Handle(ShareRequest request, CancellationToken cancellationToken)
+            {
+                var reason = request.Reason;
+                if (string.IsNullOrEmpty(reason))
+                {
+                    reason = $"Shared by {request.Editor.Person.Name}";
+                }
+
+                var existingRequests = await db.SharedRequests
+                    .Where(x => x.RequestId == request.RequestId && x.Source == request.Source)
+                    .ToDictionaryAsync(x => x.SharedWithId, cancellationToken);
+
+                foreach (var sharedWith in request.SharedWith)
+                {
+                    var person = await profileService.EnsurePersonAsync(sharedWith);
+                    if (existingRequests.TryGetValue(person!.Id, out var existingRequest))
```

Comment:
What if person not found? Just ignore ?

Reply:
The validator for the request checks that persons exist. If person is null here, I think it's ok to throw a NRE.